python/lsst/pipe/base/task.py

```diff
@@ -155,12 +160,16 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
                 config = self.ConfigClass()
             self._taskDict = dict()
             loggerName = self._fullName
-            if log is not None and log.getName():
-                loggerName = log.getName() + '.' + loggerName
+            if log is not None and (name := log.name):
```

Comment:
Is the walrus really worth it here?  Seems like it adds 10 characters here to save 4 characters on the next line; the extra property lookup shouldn't be terrible.

Reply:
In fact I've realized we should be doing this:

```python
if log is not None and log.name:
    loggerName = log.getChild(loggerName).name
```
and may be simpler to just do:
```python
self.log = getTaskLogger(name, log)
```